packages/core/src/common/debounce.spec.ts

```diff
@@ -0,0 +1,28 @@
+/********************************************************************************
+ * Copyright (C) 2018 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { expect } from 'chai';
+import { debounceAsync } from './promise-util';
+
+describe('debounceAsync', () => {
+
+    it('debounceAsync should return a promise', async () => {
+        const promise = await debounceAsync(
+            async () => 2 + 2, 50)();
```

Comment:
The name `promise` is likely incorrect here, since it is `awaited` it is no longer a promise (as it has been resolved).